### PR TITLE
Transaction sort logic 2

### PR DESCRIPTION
**Pull request by miiu96**
- Extended the transaction from the `outport` package with an extra field `ExecutionOrder`.
- Added new fields in the `outport.Pool` structure about scheduled execution.

### CODE REVIEW

File: core/transaction/transactionSorter.go [hunk 1]
```diff
@@ -34,6 +34,34 @@ func SortTransactionsBySenderAndNonceWithFrontRunningProtection(transactions []d
 	sort.Slice(transactions, sorter)
 }
 
+// TODO remove duplicated function when will use the new version of go
+
+// SortTransactionsBySenderAndNonceWithFrontRunningProtectionExtendedTransactions - sorts the transactions by address and randomness source to protect from front running
+func SortTransactionsBySenderAndNonceWithFrontRunningProtectionExtendedTransactions(transactions []data.TransactionHandlerWithGasUsedAndFee, hasher hashing.Hasher, randomness []byte) {
```
Comment (iulianpascalau) [inline core/transaction/transactionSorter.go:40]: can you also bring the unit tests?

Comment (iulianpascalau) [inline core/transaction/transactionSorter.go:40]: Looks the same as in elrond-go :+1:

Comment (SebastianMarian) [reply core/transaction/transactionSorter.go:40]: +1

Comment (miiu96) [reply core/transaction/transactionSorter.go:40]: extended the current unit tests

**Review by SebastianMarian — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by SebastianMarian — DISMISSED** [review]

File: core/transaction/transactionSorter.go [hunk 2]
```diff
@@ -34,6 +34,34 @@ func SortTransactionsBySenderAndNonceWithFrontRunningProtection(transactions []d
 	sort.Slice(transactions, sorter)
 }
 
+// TODO remove duplicated function when will use the new version of go
```
[the remark this answers is not in the document]
Comment (miiu96) [reply core/transaction/transactionSorter.go]: done

**Review by iulianpascalau — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**